packages/mui-material/src/internal/SwitchBase.test.js

```diff
@@ -442,7 +442,7 @@ describe('<SwitchBase />', () => {
         reactMajor >= 19 && 'A component is changing a controlled input to be uncontrolled.',
         reactMajor < 19 &&
           reactMajor !== 16 &&
-          'Warning: A component is changing an uncontrolled input to be controlled.',
+          'Warning: A component is changing a controlled input to be uncontrolled.',
```

Comment:
This message is correct according to what the test does. No idea why it passes for other react versions